woocommerce/wc-admin
# Forward `autocompleter` prop from `CompareFilter` to `Search`

_(Follow-up to https://github.com/woocommerce/woocommerce-admin/pull/6880, to make `ReportFilters support custom Search)_

- Forward `autocompleter` prop from `CompareFilter` to `Search`.
Allow, to use the `custom` type of search, previously it was complaining about lack of `autocompleter`, even though it was provided.

- Move `path` in Storybooks example to a parameter, to allow setting it in unit tests.
- Add few tests for FilterPicker. 
	- it renders the basic storybook example without throwing an error
	- it forwards `autocompleter` & `type` props

Fixes: woocommerce/woocommerce-admin#6890
Needed for:242-gh-woocommerce/google-listings-and-ads

### Accessibility




No new texts or animations were introduced, existing ones behave as before.
-   [n/a] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
-   [n/a] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

### Screenshots

### Detailed test instructions:

-   Create `CompareFilter` with `type="custom"` and `autocompleter`
	```jsx
	<CompareFilter
		type: 'custom',
		autocompleter: {...} // Even the one copied from exiting ones.
	/>
	``` 
-   Check for errors.





#### Doubts:
1. I was wondering whether maybe the more sustainable and flexible solution would be to either forward all the props to allow customization of the Search element, and do not curate the list of forwarded ones.
